src/main/java/com/teragrep/cnf_01/DeepCopiedMap.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Teragrep Java Configuration Library (cnf_01)
+ * Copyright (C) 2024 Suomen Kanuuna Oy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.cnf_01;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * An immutable deep copy of a Map.
+ */
+public final class DeepCopiedMap {
+
+    private final Map<String, String> map;
+
+    public DeepCopiedMap(final Map<String, String> map) {
+        this.map = map;
+    }
+
+    public Map<String, String> map() {
+        return Collections
+                .unmodifiableMap(map.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
```

Comment:
this should be done in the constructor, because otherwise the map would not be a deepcopied copy but a normal modifiable copy and result into changes in values to map() call.

Reply:
this is because Map object does not provide toImmutable() which would return this kind of an object, of course a decorator to a map that would add this functionality might be a way to go.

Reply:
Introduced a decorator for a Map in the newest commit that has toImmutable() function. The objects name is a bit weird: `ImmutabilitySupportedMap`. Tried to think something like "ImmutingMap" but there is no word for that in English language I suppose.

To enforce type-safety, it was important that ImmutableMap object can only be created with the new decorator. For this reason, I took "public" out of the class definition to make ImmutableMap package-private. Now it can only be created with the decorator when using this library.

I added mutations for the original map in `DefaultConfigurationTest` for the test that produces the default config from the method call. The default configs can't now be mutated between the constructor call (`new DefaultConfiguration()`) and the function call (`.asMap()`).